src/include/parser/expression/subquery_expression.h

```diff
@@ -37,6 +37,52 @@ class SubqueryExpression : public AbstractExpression {
    */
   std::shared_ptr<parser::SelectStatement> GetSubselect() { return subselect_; }
 
+  void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
+
+  /**
+   * @return Derived depth of the expression
+   */
+  int DeriveDepth() override {
+    int current_depth = this->GetDepth();
+    for (auto &select_elem : subselect_->GetSelectColumns()) {
+      int select_depth = select_elem->DeriveDepth();
+      if (select_depth >= 0 && (current_depth == -1 || select_depth < current_depth)) this->SetDepth(select_depth);
+    }
+    auto where = subselect_->GetSelectCondition();
+    if (where != nullptr) {
+      auto where_depth = where->DeriveDepth();
+      if (where_depth >= 0 && where_depth < current_depth) this->SetDepth(where_depth);
```

Comment:
What Depth are we supposed to find here? The smallest depth? the variable `current_depth` is never updated, so in the end, the depth will be set to just the last depth that is smaller than the `current_depth` we stored in `current_depth`. 

I think if this code is working as intended, there should be a comment explaining what depth we want in the end.

Reply:
Depth will be set by the binder. So before expressions get into binder,  depth is never touched.

Reply:
I see.

Does this code work as intended though? In AbstractExpression, it seems you calculate the [min depth](https://github.com/cmu-db/terrier/pull/433/files/a46ea0fb46997f6182be0a9ef60e6e49609c2ec5?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&hide-deleted-files=true#diff-a5c29509067f28d9df138208f3d05ff2R182), but the above code doesn't do that since you never update `current_depth` after calling `SetDepth`